### ISSUE TITLE
`azurerm_key_vault_key` support `oct-HSM`

### ISSUE THREAD

**Issue by xli1996** [hashicorp/terraform-provider-azurerm#25801]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

Can `azurerm_key_vault_key` be used with `azurerm_key_vault_managed_hardware_security_module` which has similar interface with `azurerm_key_vault`. ~~There is an addtional key type `AES-HSM` that currently is not supported with `azurerm_key_vault_key`.~~ Actually the key type is `oct-HSM` which is removed from https://github.com/hashicorp/terraform-provider-azurerm/pull/18608.

I wonder if there should be an extra resource like `azurerm_key_vault_managed_hardware_security_module_key`? Or we can use existing resources `azurerm_key_vault_key` but support more key types?

### New or Affected Resource(s)/Data Source(s)

azurerm_key_vault_key

### Potential Terraform Configuration

```hcl
resource "azurerm_key_vault_key" "key" {
  name         = var.key_name
  key_vault_id = azurerm_key_vault_managed_hardware_security_module.vault.id
  key_type     = "oct-HSM"
  key_size     = 256
  key_opts = var.key_opts
  # rotate after 90 days
  rotation_policy {
    automatic {
      time_before_expiry = "P30D"
    }
    notify_before_expiry = "P30D"
    expire_after         = "P120D"
  }
}
```

**Comment by aristosvo** [comment]
Hi @xli1996!

Thanks for opening this issue. I was the one who removed the support for `oct-HSM` in #18608, as I couldn't make the tests work for it and documentation not showing anything about it.

Can you explain how this would work for you, with some accompanying `az` CLI or `azapi` examples of how you provision it now?

**Comment by xli1996** [comment]
@aristosvo 

Hi, thanks for the response.

I mostly followed this documentation https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault_managed_hardware_security_module and provisioned a hsm.
After that I had to manually [activate](https://learn.microsoft.com/en-us/azure/key-vault/managed-hsm/quick-create-cli#create-a-managed-hsm) that as hsm requires to upload at least three RSA certs and download one security domain. That step can probably be replaced with local exec but I manually did this step.

After proper role assignments with local RBAC then I have to use azure cli to create `oct-HSM` keys and rotation policy like
```
## Create keys and set rotation policy
az keyvault key create \
    --hsm-name $VAULT_NAME \
    --name $KEY_NAME \
    --size 256 \
    --kty oct-HSM \
    --ops decrypt encrypt sign verify unwrapKey wrapKey

az keyvault key rotation-policy update \
    --hsm-name $VAULT_NAME \
    --name $KEY_NAME \
    --value $POLICY_NAME
```
Since it is not supported in the current providers.

Let me know if you need additional information.

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #25935. As such, I am going to mark this issue as closed. If that is not the case, please provide additional information, thanks!

**Comment by rcskosir** [comment]
Thank you @jack1902 for catching that, I will reopen this issue. I will see what I can find out regarding `oct-HSM` and if it was omitted intentionally due to technical issues or if it can be added in. Due to the time of year I want to apologize in advance for the delay I may have in getting an update.
